google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -394,8 +394,11 @@ protected static String getAppEngineProjectId() {
       String serviceAccountName = (String) method.invoke(appIdentityService);
       int indexOfAtSign = serviceAccountName.indexOf('@');
       return serviceAccountName.substring(0, indexOfAtSign);
-    } catch (Exception ignore) {
-      // return null if can't determine
+    } catch (Exception exception) {
+      if (System.getProperty("com.google.appengine.runtime.version") != null) {
+        // Could not resolve appengine classes under GAE environment.
+        throw new RuntimeException("Unable to resolve appengine classes. Missing dependency.");
```

Comment:
The only place I found for trouble shooting section is here:
https://github.com/GoogleCloudPlatform/gcloud-common/blob/master/troubleshooting/readme.md#troubleshooting

Do you want me to add something there? The other approach is to provide more information in the error message such as "Did you forget to add the appengine-api-1.0-sdk dependency to your project?"